examples/bri-3/src/bri/workgroup/worksteps/persistence/workstepsStorage.agent.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../../../../../prisma/prisma.service';
+import { Workstep } from '../models/workstep';
+
+// Storage Agents are the only places that talk the Prisma language of models.
+// They are always mapped to and from domain objects so that the business layer of the application
+// does not have to care about the ORM.
+@Injectable()
+export class WorkstepStorageAgent extends PrismaService {
+  async getWorkstepById(id: string): Promise<Workstep> {
+    const workstepModel = await this.workstep.findUnique({ where: { id: id } });
```

Comment:
these is bug here if workstep is not found, i assume prisma will return null or undefined so line 13 will read from undefined

also on place you use this method, you assume this one will return undefined if not found
```
const workstep = await this.storageAgent.getWorkstepById(query.id);

    if (!workstep) {
      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
    
```

Reply:
Ah, good catch. I thought i covered with tests on my branch but now i realized that storageAgent is mocked. @Kasshern please also fix in my agent when you fix here.

Reply:
Removed the check for undefined and null form the query handled and placed inside the storageAgents for workstep and bpiSubject. Let me know if you would like the second layer of validation in the QueryHandler to remain - seems redundant. @ognjenkurtic @skosito